Remove core/post-comments-form block styles that are overriding button element styles

matiasbenedetto opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
Remove core/post-comments-form block styles that are overriding button element styles from `theme.json`

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

These styles are preventing that the `theme.json` styles for button elements apply over the core/post-comments-form submit button.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
I think those styles can be removed because they are no longer needed and they are conflicting with the button element styles.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

1. Add theme.json settings for core/button block:
```
	"styles": {
              "elements": {
                      "button": {
                              "border": {
	                              "color": "black",
	                              "style": "solid",
	                              "width": "3px"
                              }
              },
	...
```
2. Check the rendered post comments form and see if the styles are being overridden.

## Screenshots or screencast <!-- if applicable -->

Using the `theme.json` from the top :arrow_up:  example:

**Before:**
![image](https://user-images.githubusercontent.com/1310626/176474203-7da7490c-a258-4b5e-9591-3b4f6ce9ee1b.png)


**After:**
![image](https://user-images.githubusercontent.com/1310626/176473877-9d8b2ea4-68b8-4b87-8fa9-aeea715da417.png)

matiasbenedetto committed:
Post comments form: removing styles targeting submit button because they are overriding the core/block button styles.

draganescu reviewed (commented):
I think we should only remove the styles that are provided by core `theme.json`. For example `cursor:pointer` is not, so it should remain. Testing this I saw no difference because the class `.wp-block-button__link` in the button block's CSS adds `cursor:pointer`, but IMO it should not be removed - same to all other CSS that are not in `theme.json`.

matiasbenedetto committed:
Removing just the properties provided by theme.json

matiasbenedetto commented:
I updated the testing instructions following the [issue conversation](https://github.com/WordPress/gutenberg/issues/42052). 
And, as @draganescu suggested, I restored all the properties but the `border` one, which can be provided by `themes.json`.

draganescu reviewed (approved):
Let's land this 👍🏻

carolinan commented:
Using Empty theme.
Screenshots from the block editor, not site editor.

**Before:**
WordPress 6.0, for comparison. Button styles in the editor and front do not match (Gutenberg is deactivated here).
![comment-form-button-gb-decativated](https://user-images.githubusercontent.com/7422055/176616687-02f195ff-1a0b-49ea-91de-583b90f64a27.png)

With Gutenberg trunk, and changes copied from PR description and added to theme.json:
![comment-form-button-gb-trunk](https://user-images.githubusercontent.com/7422055/176617749-e23a33da-7fea-479c-8cce-077227598acb.png)

**After, with PR applied:**
Changes copied from PR description and added to theme.json:
![comment-form-button-pr-json-changes](https://user-images.githubusercontent.com/7422055/176619469-382a2f8e-0956-47fa-be62-7ce307f466a2.png)

Without button styles in theme.json, the button has the browser default border for buttons (and I assume this is the style that will be the default style used in classic themes).

![comment-form-button-has-default-border](https://user-images.githubusercontent.com/7422055/176619873-2313eedf-2b34-48ff-8b49-fccc59bd6d82.png)


----

There is also an issue with the border radius, but I do not know which of the button element PR's that caused it. I suspect that this might even bleed through from the WordPress admin area styles, from the .submit CSS class?

carolinan commented:
I am not fond of the browser default border that will be used if theme authors do not style this button, but I also do not have a solution for it.

draganescu reviewed (dismissed):
Let's wait on figuring out what's up with the roundness in @carolinan 's tests

scruffian committed:
Elements: Buttons: Style the cursor for all button elements

scruffian committed:
move to block library

scruffian committed:
move to style.scss

carolinan commented:
The `.submit` style has a higher specificity.
I assume the submit class was added at some point to match the original comment_form() function.

![image](https://user-images.githubusercontent.com/7422055/176993620-85673c21-4d58-4556-92c4-4792413bc959.png)

matiasbenedetto committed:
Element styles: Load them in the common.scss file instead of the style.scss file to make them available on the frontend.

matiasbenedetto committed:
Merge branch 'trunk' into fix/button-cursors-scss-path

matiasbenedetto committed:
Removing elements.scss from style.scss

matiasbenedetto committed:
Merge branch 'fix/button-cursors-scss-path' into fix/42052

matiasbenedetto committed:
core/post-comments-form: removing unneeded class from submit button to avoid unwanted styles

matiasbenedetto committed:
default theme.json button element settings: adding border 0.

matiasbenedetto commented:
The problems around border-radius found by @carolinan preexist this PR and they are not strictly related to the original intention of this PR. 
![image](https://user-images.githubusercontent.com/1310626/177388536-a2d290f3-e083-4171-b8dc-bf1c08afd564.png)
Despite that, I introduced a commit removing the `.submit` class introduced [in this PR](https://github.com/WordPress/gutenberg/pull/41251). Removing that class doesn't seem to be affecting the form button and fixes the `border-radius` problem which caused the weird button shape :point_up: .

Apart from that, I added a default setting in Gutenberg's default `theme.json` to remove the user-agent default button border. In this way, we are able to style the `core/post-comments-form` submit button using the theme's `theme.json` element button (`styles.elements.button`).

In this screenshot, you can see that both problems seem to be solved. For this example, I'm using `emptytheme` and these settings in the theme's `theme.json`.

```
	"styles": {
		"elements": {
			"button": {
				"border": {
					"color": "red",
					"style": "solid",
					"width": "10px"
				},
				"color": {
					"background": "orange",
					"text": "blue"
				}
			}
		}
	}	
```

![image](https://user-images.githubusercontent.com/1310626/177389385-bff752a5-467a-4bb2-9f22-921ba499ed1a.png)


With Gutenberg plugin disabled still are some visual discrepancies between the editor and the frontend caused by some admin CSS definitions. I may have a fix for that, I'm testing it. Due it is a core-problem that needs to be fixed in core repo I think we can move forward with this PR. 

Thanks @carolinan and @draganescu for your feedback!

scruffian commented:
> removing the .submit class introduced https://github.com/WordPress/gutenberg/pull/41251

I think that class pre-dates that PR...

draganescu commented on packages/block-library/src/post-comments-form/form.js:
```diff
@@ -34,7 +34,6 @@ const CommentsForm = () => {
 						name="submit"
 						type="submit"
 						className={ classnames(
-							'submit',
```
Where else is this class? If it's not needed should we remove the directives as well?

matiasbenedetto replied:
I haven't found the class anywhere else in Gutenberg codebase.
The CSS directives live in the WordPress core repo: https://github.com/WordPress/wordpress-develop/blob/0a17a80bccd452b91c3b63f71f010ba131a2c954/src/wp-admin/css/forms.css#L415-L421

matiasbenedetto committed:
Merge branch 'trunk' into fix/42052

michalczaplinski reviewed (approved):
This looks good to me 👍 

I have originally added the `submit` classes in 08d34f36ff2ebd35bb5dbdac63f26be5b76c8fb6 when creating the Post Comments Form for the editor. The only reason for adding it was because the PHP (non-block) comments also include it.

michalczaplinski reviewed (changes requested):
Uh, I'm sorry for having to revoke my approval 😅 

I have noticed that the `submit` class is also applied on the frontend through the `comment_form_defaults` filter: https://github.com/WordPress/gutenberg/blob/c8b7177d4a672a3e0de3f7f2232f15ed4e15d78b/packages/block-library/src/post-comments-form/index.php#L32

[This](https://developer.wordpress.org/reference/functions/comment_form/#source) is where the `comment_form_defaults` filter is being applied in Core.

This filter was introduced to the Post Comments Form block in https://github.com/WordPress/gutenberg/pull/40628 because of the feedback in https://github.com/WordPress/gutenberg/issues/34994#issuecomment-1109754877.

So, I think we should either remove the `submit` class in both places or keep it in both 🙂 .

matiasbenedetto committed:
post comments form server side rendering: removing the 'submit' css class

matiasbenedetto commented:
Thanks for your detailed feedback @michalczaplinski !
In [the last commit](https://github.com/WordPress/gutenberg/pull/42053/commits/4912608ee39daeb16739448746f907e61b94e5ed), I removed the `submit` class also from the server-side rendering.  Do you think that could work?

michalczaplinski reviewed (approved):
I think that's fine! 🙂 

There is only one class (`submit`) applied to the button which is defined [here](https://github.com/WordPress/wordpress-develop/blob/3a518382cb455756b81961bfb558f0e7a012eee7/src/wp-includes/comment-template.php#L2491) and added to the final markup [here](https://github.com/WordPress/wordpress-develop/blob/3a518382cb455756b81961bfb558f0e7a012eee7/src/wp-includes/comment-template.php#L2693)

Also, I wanna note [this comment](https://github.com/WordPress/gutenberg/discussions/35470#discussioncomment-1450059) which confirms that CSS classes can change from time to time given good reasons (I'm paraphrasing).
